File: piece-of-cake/piece-of-cake.js
"use strict";
process.stdin.resume();
process.stdin.resume();
let inputString = "";
let currentLine = 0;
process.stdin.on("data", inputStdin => {
    inputString += inputStdin;
});
process.stdin.on("end", _ => {
    inputString = inputString.trim().split("\n").map(string => {
        return string.trim();
    });
    main();
});
function readLine() {
    return inputString[currentLine++];
}
function main() {
    const h = 4;
    const values = readLine().split(" ");
    const w = parseInt(values[0]);
    const horizontalCut = parseInt(values[1]);
    const verticalCut = parseInt(values[2]);

    const qValues = [];
    qValues.push(horizontalCut * verticalCut * h);
    qValues.push((w - horizontalCut) * verticalCut * h);
    qValues.push((w - horizontalCut) * (w - verticalCut) * h);
    qValues.push(horizontalCut * (w - verticalCut) * h);

    console.log(Math.max(...qValues));
}

File: piece-of-cake/piece-of-cake.py
import sys

input = sys.stdin.readline().rstrip().split(' ')
h = 4
w = int(input[0])
horizontalCut = int(input[1])
verticalCut = int(input[2])

qValues = []
qValues.append(horizontalCut * verticalCut * h)
qValues.append((w - horizontalCut) * verticalCut * h)
qValues.append((w - horizontalCut) * (w - verticalCut) * h)
qValues.append(horizontalCut * (w - verticalCut) * h)

print(max(qValues))